packages/semcom-upload-component/lib/components/sem-com-store-selection.component.ts

```diff
@@ -0,0 +1,220 @@
+import { html, unsafeCSS, css, CSSResult, TemplateResult, property, state, query } from 'lit-element';
+import { Theme } from '@digita-ai/ui-transfer-theme';
+import { RxLitElement } from 'rx-lit';
+import { DoneEvent, interpret } from 'xstate';
+import { FormCleanlinessStates, FormRootStates, FormSubmissionStates, FormValidationStates, FormEvents, FormUpdatedEvent, formMachine, FormContext, FormValidatorResult } from '@netwerk-digitaal-erfgoed/solid-crs-components';
+import { from } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export class SemComStoreSelectionComponent extends RxLitElement {
+
+  /** A space separated list of urls, each of which points to a SemCom store */
+  @property({ type: String })
+  public semComStoreUrls: string;
+
+  formMachine = formMachine<{ dropDown: string; freeInput: string }> (
+    (context) => this.validateStoreSelectionForm(context)
+  ).withContext({ data: { dropDown: 'empty', freeInput: '' }, original: { dropDown: 'empty', freeInput: '' } });
+
+  /** The actor responsible for form validation in this component.  */
+  formActor = interpret(this.formMachine, { devTools: true });
+
+  /** Indicates if if the form validation passed. */
+  @state()
+  isValid = false;
+
+  /** Indicates if one of the form fields has changed. */
+  @state()
+  isDirty = false;
+
+  /** The freeInput element from the DOM */
+  @query('#freeInput')
+  freeInput: HTMLInputElement;
+
+  /** The dropDown element from the DOM */
+  @query('#dropDown')
+  dropDown: HTMLSelectElement;
+
+  constructor() {
+
+    super();
+
+    this.subscribe('isValid', from(this.formActor).pipe(
+      map((machineState) => machineState.matches({
+        [FormSubmissionStates.NOT_SUBMITTED]:{
+          [FormRootStates.VALIDATION]: FormValidationStates.VALID,
+        },
+      })),
+    ));
+
+    this.subscribe('isDirty', from(this.formActor).pipe(
+      map((machineState) => machineState.matches({
+        [FormSubmissionStates.NOT_SUBMITTED]:{
+          [FormRootStates.CLEANLINESS]: FormCleanlinessStates.DIRTY,
+        },
+      })),
+    ));
+
+    this.formActor.onDone((event: DoneEvent) => { this.dispatchEvent(new CustomEvent('formSubmitted', { detail: { dropDown: event.data.data.dropDown, freeInput: event.data.data.freeInput } })); });
+
+    this.formActor.start();
+
+  }
+
+  validateStoreSelectionForm =
+  async (context: FormContext<{ dropDown: string; freeInput: string }>): Promise<FormValidatorResult[]> => {
+
+    const res: FormValidatorResult[] = [];
+
+    // only validate dirty fields
+    const dirtyFields = Object.keys(context.data).filter((field) =>
+      context.data[field as keyof { dropDown: string; freeInput: string }]
+    !== context.original[field as keyof { dropDown: string; freeInput: string }]);
```

Comment:
Yes, this is copied from there. Not typing it like this gives error:

```Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{ dropDown: string; freeInput: string; }'.
  No index signature with a parameter of type 'string' was found on type '{ dropDown: string; freeInput: string; }'.```

Reply:
fixed by adding index signature to the contexts: '[key: string]: string;'